### PR TITLE
nc_coord_var works for a lot more data now

### LINKED ISSUE #9: do Coordinates properly
http://cfconventions.org/Data/cf-conventions/cf-conventions-1.7/cf-conventions.html#coordinate-system

Also see discussion and examples here: https://github.com/r-spatial/stars/issues/62#issuecomment-426991912

### PR DESCRIPTION
**Pull request by dblodgett-usgs**
This should fix #9 

This runs for everything in rasterwise and I pulled in three more files that were causing problems to make sure the edge cases are well tested. 

I'll make a few comments in line to clarify why this does what it does...

### CODE REVIEW

File: R/nc_coord.R [hunk 1]
```diff
@@ -58,9 +58,11 @@ nc_coord_var_call <- function(dim, var, att, axe, variable) {
   }
 }
 
-#' @importFrom dplyr bind_rows filter select left_join group_by arrange mutate ungroup
+#' @importFrom dplyr bind_rows filter select left_join group_by arrange mutate ungroup distinct
 nc_coord_var_finder <- function(dim, var, att, axe, variable) {
   
+  if(nrow(att) == 0) return(NULL)
```
Comment (dblodgett-usgs) [inline R/nc_coord.R]: This just kills off warnings in the case that there are no attributes to work with.

File: R/nc_coord.R [hunk 2]
```diff
@@ -79,60 +81,67 @@ nc_coord_var_finder <- function(dim, var, att, axe, variable) {
...
   
   coord_var <- sapply(coord_vars, divine_XYZT,
                       atts = filter(att, variable %in% coord_vars), 
                       simplify = FALSE)
   
-  coord_var <- replace(coord_var, sapply(coord_var, is.null), "unknown")
-  
-  coord_var_base <- tibble::as_tibble(list(coord_var = names(coord_var),
-                                      axis = unlist(coord_var)))
+  coord_var <- coord_var[!sapply(coord_var, is.null)]
```
Comment (dblodgett-usgs) [inline R/nc_coord.R]: Nulls mixed in were causing problems. Switched from "unknown" to just not including them. See comments in #9 for discussion of why.

File: R/nc_coord.R [hunk 3]
```diff
@@ -79,60 +81,67 @@ nc_coord_var_finder <- function(dim, var, att, axe, variable) {
...
-  } else {
-    bind_rows(out, coordinates_atts %>%
-                select(variable, value) %>%
-                mutate(value = as.character(value)) %>%
-                mutate(value = strsplit(value, " ")) %>%
-                tidyr::unnest() %>%
-                left_join(coord_var_base, by = c("value" = "coord_var")) %>%
-                tidyr::spread(key = axis, value = value)
-    )
+    coord_var <- tibble::as_tibble(list(variable = variable)) %>%
```
Comment (dblodgett-usgs) [inline R/nc_coord.R]: These changes account for the need to simultaneously support 1d dimension pair (COARDS style) coordinate variables and potentially 2d CF auxiliary coordinate variables. Seems to work pretty well.

**Comment by mdsumner** [comment]
Very nice! Here's a couple of very scary files tested ;) - the second one is way to many rows to show!

image


image

**Pull request merged**